dipy/viz/window.py

```diff
@@ -591,6 +586,28 @@ def add_window_callback(self, win_callback):
         self.window.AddObserver(vtk.vtkCommand.ModifiedEvent, win_callback)
         self.window.Render()
 
+    def add_timer_callback(self, repeat, duration, timer_callback):
+        self.iren.AddObserver("TimerEvent", timer_callback)
```

Comment:
Rather than using `timer_callback` directly, we could wrap it with our own method/function that way we can control what arguments will be provided to `timer_callback`.

For instance, at the moment `timer_callback` is a function that expects two arguments: a `vtkRenderWindowInteractor` and a `str` (the event's name). I doubt common users would use them but they are still forced to provide a callback function that accepts those two arguments. What I propose is the following:

```python
def _wrap_timer_callback(self, timer_callback):
    def _wrapped_timer_callback(iren, event):
        timer_callback(self, ...)

   return _wrapped_timer_callback
```
then replace line 590 with
```python
self.iren.AddObserver("TimerEvent", self._wrap_timer_callback(timer_callback))
```
Note `...` means we could provide any Python object we deem relevant in addition to the `ShowManager` instance (i.e. `self`).

Reply:
Okay, this is an important point. But does not make much sense changing that in this PR. We need to make decisions about callbacks in a developers meeting. I know that each one of us @karandeepSJ, @dmreagan, @ranveeraggarwal and myself have different suggestions of what will be the optimal strategy with callbacks.  My plan is to have that discussion as soon as possible. Possibly after the first split of viz modules. Will send a doodle about that. But for now I suggest moving ahead with this PR. Let me know if you disagree.